### DISCUSSION TITLE
Why setting default temperature to 1.0?

### DISCUSSION THREAD

**Discussion by jinliu** [TheR1D/shell_gpt#97] [General]
GPT seems quite likely to make wrong answers in this setting.

For example, 
`sgpt --no-cache --shell "docker update all images"`
often produces
`docker image update --all`
where there's no "update" command at all; and
`docker images -q | xargs docker pull`
which is wrong.

Meanwhile
`sgpt --no-cache --shell --temperature=0 "docker update all images"`
produces
`docker images | awk '{print $1}' | xargs -L1 docker pull`
which is almost right (should exclude the first line).

So in such high temperature, GPT may produce wrong answers in first run, when it actually "knows" the right answer.

I think users run sgpt mainly for most correct answers (low temperature), not for creativity (high temperature). Maybe we default with temperature=0, and when the user run the same query repeatedly (meaning he's not satisfied with the answer), we gradually raise the temperature to get more answers?

**Comment by lpurdy01** [comment]
I think just because 1.0 is fun.

**Comment by TheR1D** [comment]
The default temperature was changed to 0.1 in one of the last patches.
